packages/angular_devkit/schematics/src/rules/utils/is-binary_spec.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { isBinary } from './is-binary';
+
+describe('isBinary', () => {
+  it('should be false if provided value is a UTF-8 buffer', () => {
+    const fooBuffer = Buffer.from('foo');
+    const jpBuffer = Buffer.from('こんにちは世界！');
+
+    expect(isBinary(fooBuffer)).toBe(false);
+    expect(isBinary(jpBuffer)).toBe(false);
+  });
+
+  it('should be true if provided value is a buffer with unknown ASCII character', () => {
+    const buffer = Buffer.from([128]);
```

Comment:
This is not correct.  128 is a valid extended ASCII character.  UTF-8 supports extended ASCII via fallback.

Reply:
```js
const { TextDecoder } = require('util');

const decoder = new TextDecoder('utf-8', { fatal: true });

console.log(decoder.decode(Buffer.from([128])));
```

```
TypeError [ERR_ENCODING_INVALID_ENCODED_DATA]: The encoded data was not valid for encoding utf-8
```

Reply:
That is because the algorithm used by `TextDecoder` does not implement fallback.

Reply:
Interesting...

Just been googling for some results about this and every answer on the StackOverflow or wherever else says like:
> JavaScript deals only in Unicode, not Extended ASCII.

Reply:
It _kind of_ deals in unicode.  This article provides some background on the overall unicode issue: https://mathiasbynens.be/notes/javascript-unicode

Reply:
The underlying issue here though is that `isBinary` shouldn't discriminate based on the text encoding method.  An additional reason why I think the new approach is more appropriate.